cloudify_cli/labels_utils.py

```diff
@@ -46,10 +48,26 @@ def get_output_resource_labels(raw_resource_labels):
 
 
 def get_printable_resource_labels(resource_labels):
-    return [
-        {'key': resource_label_key, 'values': resource_label_values} for
-        resource_label_key, resource_label_values in resource_labels.items()
-    ]
+    printable_labels = []
+    for resource_label_key, resource_label_values in resource_labels.items():
+        formatted_label_values = [_format_label_value(label_value) for
+                                  label_value in resource_label_values]
+        printable_labels.append({'key': resource_label_key,
+                                 'values': formatted_label_values})
+    return printable_labels
+
+
+def _format_label_value(label_value):
+    label_value = label_value.replace(',', '\\,').replace(':', '\\:').\
+        replace('$', '\\$')
+    if label_value_needs_quotes(label_value):
```

Comment:
Not sure why not just quote any label value, for uniformity's sake

Reply:
Agreed.